os/fs/smartfs/smartfs_smart.c

```diff
@@ -1043,7 +1049,7 @@ static int smartfs_opendir(struct inode *mountpt, const char *relpath, struct fs
 	/* Search for the path on the volume */
 
 	entry.name = NULL;
-	ret = smartfs_finddirentry(fs, &entry, relpath);
+	ret = smartfs_finddirentry(fs, &entry, relpath, false);
```

Comment:
so here, entry type is directory, we don't have to get datalen.

Reply:
Wherever we have functions related to directories, we do not need to calculate data length.
Also, in such cases it does not matter because the flags will be `SMARTFS_DIRENT_TYPE_DIR`, so anyways file's data length is not calculated in old version also.